subcommand/server-acl-init/policies.go

```diff
@@ -0,0 +1,208 @@
+package serveraclinit
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+)
+
+type rulesData struct {
+	EnableNamespaces    bool
+	ConsulSyncNamespace string
+	EnableNSMirroring   bool
+	MirroringPrefix     string
+}
+
+// Rules that don't currently have any
+// modifications based on config changes
+func (c *Command) snapshotAgentRules() string {
+	snapshotAgentRules := `acl = "write"
+key "consul-snapshot/lock" {
+   policy = "write"
+}
+session_prefix "" {
+   policy = "write"
+}
+service "consul-snapshot" {
+   policy = "write"
+}`
+
+	return snapshotAgentRules
+}
+
+func (c *Command) entLicenseRules() string {
+	entLicenseRules := `operator = "write"`
+
+	return entLicenseRules
+}
+
+// Rules that depend on namespace configuration
+func (c *Command) agentRules() (string, error) {
+	templateData := rulesData{
+		EnableNamespaces: c.flagEnableNamespaces,
+	}
+
+	agentRulesTpl := `
+  node_prefix "" {
+    policy = "write"
+  }
+{{- if .EnableNamespaces }}
+namespace_prefix "" {
+{{- end }}
+  service_prefix "" {
+    policy = "read"
+  }
+{{- if .EnableNamespaces }}
+}
```

Comment:
I'm finding this pretty hard to read and I'm worried it's fragile to try and get the braces matching with the conditionals. What do you think about duplicating it for clarity:
```
node_prefix "" {
  policy = "write"
}
{{- if .EnableNamespaces }}
namespace_prefix "" {
  service_prefix "" {
    policy = "read"
  }
}
{{- else }}
service_prefix "" {
  policy = "read"
}
{{- end }}

```

Reply:
There's a couple of methods here where I think this method would make it easier to read.

Reply:
Whitespace is ignored in these rules. It ends up being a lot of duplication and makes the templates be overkill except for catalog sync.